File: medium/container_with_most_water.py
import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        l = 0
        r = len(height)-1
        max = 0
        while True:
            if height[l] > height[r]:
                if max < height[r]*(r-l):
                    max = height[r]*(r-l)
                r = r-1
            else:
                if max < height[l]*(r-l):
                    max = height[l]*(r-l)
                l = l+1
            if l==r:
                break
        return max

File: medium/container_with_most_water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function (height) {
  let l = 0;
  let r = height.length - 1;
  let max = 0;
  while (true) {
    if (height[l] > height[r]) {
      if (height[r] * (r - l) > max) {
        max = height[r] * (r - l);
      }
      r = r - 1;
    } else {
      if (height[l] * (r - l) > max) {
        max = height[l] * (r - l);
      }
      l = l + 1;
    }

    if (l === r) {
      break;
    }
  }
  return max;
};
